repository/http/src/main/java/org/eclipse/rdf4j/repository/http/HTTPRepositoryConnection.java

```diff
@@ -368,12 +368,16 @@ public void rollback()
 	public void close()
 		throws RepositoryException
 	{
-		if (isActive()) {
-			logger.warn("Rolling back transaction due to connection close", new Throwable());
-			rollback();
+		try {
+			if (isActive()) {
+				logger.warn("Rolling back transaction due to connection close", new Throwable());
+				rollback();
+			}
+			super.close();
```

Comment:
If rollback fails with an exception, the client will be closed below, ``but super.close()`` will not be called